eXo/Sources/Controllers/Setting/ServerEditViewController.swift

```diff
@@ -0,0 +1,157 @@
+                        if let platformVersion = json["platformVersion"].string {
+                            let version = (platformVersion as NSString).floatValue
+                            if (version >= Config.supportVersion){
+                                self.server.serverURL = serverURL
+                                ServerManager.sharedInstance.saveServerList()
+                                NSOperationQueue.mainQueue().addOperationWithBlock({ () -> Void in
+                                    self.navigationController?.popViewControllerAnimated(true)
+                                })
+                                
+                            } else {
+                                // this application supports only platform version 4.3 or later
+                                NSOperationQueue.mainQueue().addOperationWithBlock({ () -> Void in
+                                    Tool.showErrorMessageForCode(ConnectionError.ServerVersionNotSupport)
+                                })
+                            }
+                        } else {
+                            NSOperationQueue.mainQueue().addOperationWithBlock({ () -> Void in
+                                Tool.showErrorMessageForCode(ConnectionError.ServerVersionNotFound)
+                            })
+                        }
+                        
+                    } else {
+                        NSOperationQueue.mainQueue().addOperationWithBlock({ () -> Void in
+                            Tool.showErrorMessageForCode(ConnectionError.URLError)
+                        })
+                    }
+                } else {
+                    NSOperationQueue.mainQueue().addOperationWithBlock({ () -> Void in
+                        Tool.showErrorMessageForCode(ConnectionError.URLError)
+                    })
+                }
+            })
+        } else {
+            Tool.showErrorMessageForCode(ConnectionError.URLError)
+        }
+    }
+    
+    //MARK : KeyBoard handle
+    // Set up the position of the delete button to a visiable position (on portrait only)
+
+    func keyboardWillShow(notification: NSNotification) {
+        if UIDeviceOrientationIsPortrait(UIDevice.currentDevice().orientation) {
+            if let keyboardSize = (notification.userInfo?[UIKeyboardFrameBeginUserInfoKey] as? NSValue)?.CGRectValue() {
+                self.view.layoutIfNeeded()
+                // Animate the movement of the deleteButton
+                UIView.animateKeyframesWithDuration(0.5, delay: 0.0, options: UIViewKeyframeAnimationOptions.LayoutSubviews, animations: { () -> Void in
+                    self.deleteButtonConstraintToBottom.constant = keyboardSize.height + 20.0
+                    self.view.layoutIfNeeded()
+                    }, completion: nil)
+            }
+        } else {
+            // Dont change the position of the delete Button in lanscape mode
+            self.deleteButtonConstraintToBottom.constant = 20.0
```

Comment:
Just like you do in `SettingViewController.swift`, you can define a constant `kDeleteButtonBottomMargin` to 20.0 and use it here.